README.md

```diff
@@ -379,6 +379,34 @@ Succeeded   : 50
 
 </details>
 
+## Design
+
+### High-level Decisions
+
+- **Nextflow Workflow**
+  - Many QC tests will require the output from one or more commands using external (_i.e._ non-Python) programs. While we can run external commands in Python (_e.g. using the `subprocess` module), we want to avoid re-inventing a workflow management system (WMS). Instead, we should leverage an existing WMS. For the purposes of this POC, we opted for Nextflow since we have the compute infrastructure set up. That said, a future iteration could use the Common Workflow Language (whose reference implementation is in Python) and/or the Task Execution Service (TES).
+  - Given that Nextflow supports arbitrary Groovy/Java code, we could technically accomplish all of the work in Nextflow alone. However, it makes a lot more sense to use a programming language that our team (_i.e._ DPE) is familiar with and that people outside of our team can readily contribute to. Python fits the bill. Hence, we are currently building a Python-Nextflow hybrid tool. 
+  - In Nextflow, the `container` directive and the `script` block can leverage values provided as inputs (_e.g._ `container "${some_input}"`). This functionality allows us to keep the Nextflow workflow very general. Otherwise, we would have to hard-code the containers and commands in the Nextflow workflow and somehow keep everything in sync with the Python package, which takes care of interpreting the commands.
```

Comment:
I like this about Nextflow. One concern I have about this is not all Docker images are supported by Tower, @BWMac also ran into an issue like [this package](https://github.com/Sage-Bionetworks/agora-data-tools/blob/40ae5870c4c6965a4931bbb6961de76ca9f63e23/Dockerfile#L4) needing to be installed.

This is tangential to this PR, but I imagine we can potentially do this for the ITCR docker execution recipe.